template/cmake-cpp/flake.nix

```diff
@@ -0,0 +1,53 @@
+{
+  description = "Example kickstart C/C++ cmake project.";
+
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+
+  outputs = inputs@{ flake-parts, ... }:
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      systems = [ "x86_64-linux" "aarch64-linux" "aarch64-darwin" "x86_64-darwin" ];
+      perSystem = { config, self', inputs', pkgs, system, ... }:
+        let
+          inherit (pkgs) dockerTools cmake;
+          inherit (dockerTools) buildImage;
+          name = "example";
+          version = "0.1.0";
+        in
+        {
+          devShells.default = pkgs.mkShell {
+            inputsFrom = [ self'.packages.default ];
+          };
+
+          packages = {
+            default = pkgs.stdenv.mkDerivation {
+              inherit version;
+              pname = name;
+              src = ./.;
+
+              buildInputs = [ cmake ];
+
+              configurePhase = ''
+                cmake .
```

Comment:
So, I don't know nix at all, I've not followed BG's journey with nix, but I've written a lot of C++ and CMake recently -- A better approach here is instead of running `cmake .` and then changing directory and running `make`, you can do something like `cmake -S . -B build/ && cmake --build build` -- this takes the top level directory and builds it in a `build/` directory, then runs the build config (in this case `make`, but `ninja` is an option as well) from within that build/ directory -- it's less commands, less set on using make in case someone wants to use an alternative in the future, and will also mean you can have the gitignore just be `build/` instead of the dozen or so lines you have -- all of that is stored inside build instead.

Reply:
Hiya. Nix builds this in a sandboxed environment and its packaging separates configure (in this case generating the makefile) and build steps into distinct tasks. That's the reason this was done. If you want some more info I'd recommend checking out the Nix documentation on Derivations :)

Reply:
Fair enough -- I wrote the command as one long string, but you could split it up, so that you aren't defining the build type but using the one set in the cmake config (by default make but I think it's only 1 line in the cmakelists.txt to change it)